addons/account/models/account_move.py

```diff
@@ -1876,23 +1876,26 @@ def _fetch_duplicate_reference(self, matching_states=('draft', 'posted')):
         self.env["account.move"].flush_model(used_fields)
 
         move_table_and_alias = SQL("account_move AS move")
-        if not moves[0].id:  # check if record is under creation/edition in UI
+        if not all(move.id for move in moves):  # check if record is under creation/edition in UI
```

Comment:
i'm wondering if we shouldn't use any instead ? and focus in the loop on moves without id ?